test/controllers/asset/other/add/OtherAnswerControllerSpec.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.asset.other.add
+
+import base.SpecBase
+import controllers.routes._
+import models.Status.Completed
+import models.WhatKindOfAsset.Other
+import pages.AssetStatus
+import pages.asset.WhatKindOfAssetPage
+import pages.asset.other.{OtherAssetDescriptionPage, OtherAssetValuePage}
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import utils.print.OtherPrintHelper
+import views.html.asset.other.add.OtherAssetAnswersView
+
+class OtherAnswerControllerSpec extends SpecBase {
+
+  val description: String = "Other asset"
+
+  lazy val otherAnswerRoute: String = routes.OtherAnswerController.onPageLoad().url
+
+  "OtherAnswer Controller" must {
+
+    "return OK and the correct view for a GET" in {
+
+      val answers =
+        emptyUserAnswers
+          .set(WhatKindOfAssetPage, Other).success.value
+          .set(OtherAssetDescriptionPage, "Other asset").success.value
+          .set(OtherAssetValuePage, 4000L).success.value
+          .set(AssetStatus, Completed).success.value
+
+      val application = applicationBuilder(userAnswers = Some(answers)).build()
+
+      val request = FakeRequest(GET, otherAnswerRoute)
+
+      val result = route(application, request).value
+
+      val view = application.injector.instanceOf[OtherAssetAnswersView]
+      val printHelper = application.injector.instanceOf[OtherPrintHelper]
+      val answerSection = printHelper(answers, provisional = true, description)
+
+      status(result) mustEqual OK
+
+      contentAsString(result) mustEqual
+        view(answerSection)(request, messages).toString
+
+      application.stop()
+    }
+
```

Comment:
Could probably do with an extra test where you add data correctly and redirect to the next page. See BusinessAnswersControllerSpec     "redirect to the next page when valid data is submitted" 

You should be able to copy that, give it a valid set of answers and mock out the call to addOtherAsset